### PR TITLE
Update CS Dependencies

### PR DESCRIPTION
**Pull request by ryuyu**
Update our ServerCommon and other C# dependencies to mitigate potential vulnerabilities.

Waiting on testing and "official" release of ServerCommon packages.

### CODE REVIEW

File: src/NuGetGallery/Web.config [hunk 1]
```diff
@@ -492,6 +492,10 @@
   </system.webServer>
   <runtime>
     <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+			<dependentAssembly>
+				<assemblyIdentity name="NuGet.Services.Contracts" publicKeyToken="31BF3856AD364E35" culture="neutral"/>
+				<bindingRedirect oldVersion="0.0.0.0-2.50.0.0" newVersion="2.50.0.0"/>
+			</dependentAssembly>
```
Comment (shishirx34) [inline src/NuGetGallery/Web.config]: nit: indentation.

File: src/DatabaseMigrationTools/DatabaseMigrationTools.csproj [hunk 2]
```diff
@@ -65,15 +62,6 @@
...
     </PackageReference>
-    <PackageReference Include="Microsoft.Extensions.Configuration">
-      <Version>1.1.2</Version>
-    </PackageReference>
-    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions">
-      <Version>1.1.2</Version>
-    </PackageReference>
-    <PackageReference Include="Microsoft.Extensions.Configuration.Binder">
-      <Version>1.1.2</Version>
-    </PackageReference>
```
Comment (shishirx34) [inline src/DatabaseMigrationTools/DatabaseMigrationTools.csproj]: Are these dependencies not required or flowing through another dependency?

Comment (ryuyu) [reply src/DatabaseMigrationTools/DatabaseMigrationTools.csproj]: They end up coming in through other dependencies. I got rid of things that we don't directly depend on as they were causing downgrade warnings. After this PR, build should completely without any NU downgrade warnings.

File: src/NuGetGallery.Core/NuGetGallery.Core.csproj [hunk 3]
```diff
@@ -256,10 +256,7 @@
       <Version>1.0.0</Version>
     </PackageReference>
     <PackageReference Include="Microsoft.Data.Services.Client">
-      <Version>5.7.0</Version>
-    </PackageReference>
-    <PackageReference Include="Newtonsoft.Json">
-      <Version>9.0.1</Version>
+      <Version>5.8.4</Version>
```
Comment (scottbommarito) [inline src/NuGetGallery.Core/NuGetGallery.Core.csproj]: Is this not pulled in through `NuGet.Services.Storage` -> `WindowsAzure.Storage`?

File: tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj [hunk 4]
```diff
@@ -327,17 +327,8 @@
...
-    </PackageReference>
-    <PackageReference Include="Microsoft.Data.OData">
-      <Version>5.7.0</Version>
-    </PackageReference>
     <PackageReference Include="Microsoft.Data.Services">
-      <Version>5.7.0</Version>
-    </PackageReference>
-    <PackageReference Include="Microsoft.Data.Services.Client">
-      <Version>5.7.0</Version>
+      <Version>5.8.4</Version>
```
Comment (scottbommarito) [inline tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj]: Is this not pulled in through a project reference on `NuGetGallery`?

**Pull request merged**